src/python/zquantum/core/measurement.py

```diff
@@ -435,28 +446,69 @@ def check_parity(
     return result
 
 
+def check_parity_of_vector(
+    bitstrings_vector: np.ndarray, marked_qubits: Iterable[int]
+) -> np.ndarray:
+    """Determine if the marked qubits have even parity for each bitstring in the given
+        vector.
+
+    Args:
+        bitstring: A 2d array of bitstrings whose size is number of bistrings * number
+            of qubits
+        marked_qubits: The qubits whose parity is to be determined.
+
+    Returns:
+        A 1d array with size equal to number of bitstrings. Each entry is 1 if an even
+        number of the marked qubits of the corresponding bitstring are in the 1 state
+        and 0 if otherwise.
+    """
+    if not marked_qubits:
+        return np.ones(bitstrings_vector.shape[0])
+
+    # Check if an even number of the marked qubits of each bitstring are in the 1 state
+    bitstring_subset = bitstrings_vector[:, np.fromiter(marked_qubits, dtype=int)]
+    return (bitstring_subset.sum(axis=1) + 1) % 2
+
+
+def _convert_bitstrings_to_vector(bitstrings: Iterable[str]) -> np.ndarray:
+    """Converts bitstrings to vector so parity can be checked."""
+    n_qubits = len([*bitstrings][0])
+    all_bits = "".join(bitstrings)
+    bitstring_1d_array = np.frombuffer(all_bits.encode("utf-8"), "u1") - ord("0")
+    return bitstring_1d_array.astype(int).reshape(-1, n_qubits)
+
+
 def get_expectation_value_from_frequencies(
     marked_qubits: Iterable[int], bitstring_frequencies: Dict[str, int]
 ) -> float:
-    """Get the expectation value the product of Z operators on selected qubits
-    from bitstring frequencies.
+    """Get the expectation values of a set of operators (with precisions) from a set of
```

Comment:
[request] I think this docstring has been edited by accident. Could you revert the changes here?

Reply:
Fixed in [19de1de](https://github.com/zapatacomputing/z-quantum-core/pull/452/commits/19de1de8df585a2f5117ce632456e343a6467a4d)